lib/cucumber/glue/snippet.rb

```diff
@@ -174,7 +174,8 @@ def result
 
         class DocString
           def append_block_parameter_to(array)
-            array << 'string'
+            parameter = format('string%<index>s', index: (array.size + 1 if array.include?('string')))
```

Comment:
Here I get your logic, but instead of giving this a different name in different circumstances, lets just use a name which will be unique.

A good example would be `doc_string`

Reply:
I agree this is better :) I will make the changes